app/services/catalog/soft_delete_restore.rb

```diff
@@ -0,0 +1,18 @@
+module Catalog
+  class SoftDeleteRestore
+    def initialize(record, restore_key)
+      @record = record
+      @restore_key = restore_key
+    end
+
+    def process
+      if @restore_key == Digest::SHA1.hexdigest(@record.discarded_at.to_s)
+        @record.undiscard
+      else
+        raise Catalog::NotAuthorized, "Wrong key to restore deleted record #{@record.id}"
```

Comment:
Possibly: `Invalid key for #{@record.id}`

I was going to say that I'm not sure I want to include the problem as part of the error message. More ambiguity would be preferable.  However we need to expose the key as part of the API spec, so the caller will know 'how it all works.'
Was wondering your thoughts @lindgrenj6

Reply:
Talked to @lindgrenj6 and realized we never actually render the message to the UI, so I am happy with this message as is.